test/geometry/liegroup/test_so3.py

```diff
@@ -124,9 +144,24 @@ def test_matrix(self, device, dtype, batch_size):
             qp_ = q1 * pquat * q1.inv()
             rp_ = torch.matmul(r1, pvec)[None, :]
             self.assert_close(rp_, qp_.vec)  # p_ = R*p = q*p*q_inv
+            self.assert_close(rp_.norm(), pvec.norm())
+
+    @pytest.mark.parametrize("batch_size", (1, 2, 5))
+    def test_ortho(self, device, dtype, batch_size):
+        q = Quaternion.random(batch_size)
+        q = q.to(device, dtype)
+        b_R_a = So3(q).matrix()
+        a_R_b = So3(q).inverse().matrix()
+        a_R_a = (So3(q) * So3(q).inverse()).matrix()
+
+        for i in range(batch_size):
+            self.assert_close(a_R_a[i, :, :], torch.eye(3))
+            self.assert_close(a_R_b[i, :, :] @ b_R_a[i, :, :], torch.eye(3))
+            self.assert_close(b_R_a[i, :, :] @ a_R_b[i, :, :], torch.eye(3))
 
```

Comment:
@stevenlovegrove can you add one extra line to validate that batched it works too?